okhttp/src/main/java/com/backblaze/b2/client/okHttpClient/B2OkHttpClientImpl.java

```diff
@@ -221,20 +223,25 @@ private String postJsonAndReturnString(String url,
     private  String postAndReturnString( String url,  B2Headers headersOrNull,  InputStream inputStream, long contentLength)
             throws B2Exception {
         RequestBody body;
+        File tempFile = null;
         try {
             if( contentLength <= (Integer.MAX_VALUE-HOTSPOT_FUDGE) ){
                 byte[] bytes = readFully(inputStream, (int) contentLength);
                 body = RequestBody.create(MediaType.get(APPLICATION_JSON), bytes);
             } else {
                 String tempFileName = UUID.randomUUID().toString();
-                File tempFile = File.createTempFile(tempFileName, ".tmp");
+                Path tempFilePath = Files.createTempFile(tempFileName, ".tmp");
+                tempFile = tempFilePath.toFile();
                 FileOutputStream outputStream = new FileOutputStream(tempFile);
                 copy( inputStream, outputStream);
                 body = RequestBody.create(MediaType.get(APPLICATION_JSON), tempFile);
-                tempFile.deleteOnExit();
             }
         } catch (IOException e) {
             throw translateToB2Exception(e, url);
+        } finally {
```

Comment:
i'm concerned that this delete is too early.
doesn't the file need to still exist during the okHttpClient.newCall() below?

Reply:
how are you testing the file-is-too-big-for-an-array code path?

Reply:
fixed